kinetic-devel release

AustinDeric opened the pull request:
addresses issue #115

Commit:
kinetic-devel release of ros-industrial/abb

gavanderhoorn reviewed (changes requested):
Nice work, three things:

 1. `industrial_core` has a Kinetic release. Is building from sources really necessary? The `abb.rosinstall` file would not be needed anymore if the released version is sufficent either.
 1. I'm not sure all of these packages need C++11 enabled, and for metapackages (`abb`) we're not allowed to change the `CMakeLists.txt`, so that will need to be reverted.
 1. the build instructions ask the user to create a `abb_experimentnal_ws` workspace, should that be `abb_ws` (also note the typo)? And if we don't use `abb.rosinstall` then the instructions should include a `rosdep install ..` step.

AustinDeric commented:
1.  Its not necessary. I will also describe the way to build this package using the Debian as well and place this before the source install method.  You will notice that the .travis.yml uses the Debian method to test.  With that said, i would prefer to leave the abb.rosinstall file for those of us who want to keep the dependencies in the local src folder.
2. I thought it may be good to use c++11 going forward, is it ok to leave it?  I will revert the metapackage.
3. typo :(. I will specify the rosdep as the "recommended" method.

Commit:
fix comments from PR #126.

gavanderhoorn commented:
> With that said, i would prefer to leave the abb.rosinstall file for those of us who want to keep the dependencies in the local src folder.

The thing is that ROS best practices are to avoid building from sources as much as possible. We can add instructions on how to build from sources, but it should be made *really* explicit that it's only *ever* needed if you want to use the packages on platforms / OS for which there is no release, you need a feature that hasn't been released yet or you want to work on the packages themselves.

> I thought it may be good to use c++11 going forward, is it ok to leave it?

I completely agree with encouraging people to use C++11, but the only packages in this repository with any C++ sources in them are the IKFast plugins and the driver. And IIRC only the IK plugins really need it enabled (because of MoveIt). So the flag doesn't do anything for the other packages, and as packages in the official repositories are taken as examples of how to set things up, it would be really nice if we can avoid adding things that don't belong. That could lead to cargo cult copy-cat behaviour which I don't think we should want.

> I will specify the rosdep as the "recommended" method.

Great. Much appreciated.

gavanderhoorn commented on README.md:
```diff
@@ -19,3 +19,30 @@ Older releases may be found in the old ROS-Industrial [subversion repository][].
+cd abb_ws/src
+git clone -b kinetic-devel https://github.com/ros-industrial/abb.git
+rosdep update
+rosdep install abb
```
May I suggest changing this to `rosdep install --from-paths . --ignore-src`?

AustinDeric replied:
fixed.

gavanderhoorn commented on README.md:
```diff
@@ -19,3 +19,30 @@ Older releases may be found in the old ROS-Industrial [subversion repository][].
+cd abb_ws/src
+git clone -b kinetic-devel https://github.com/ros-industrial/abb.git
+wstool init .
+wstool merge abb/abb.rosinstall
```
The above three lines could be simplified to read:
```
wstool init .
wstool merge https://raw.githubusercontent.com/ros-industrial/abb/kinetic-devel/abb.rosinstall
```

AustinDeric replied:
fixed.

gavanderhoorn commented on README.md:
````diff
@@ -19,3 +19,30 @@ Older releases may be found in the old ROS-Industrial [subversion repository][].
+wstool merge abb/abb.rosinstall
+wstool update
+catkin build
+```
````
And it's certainly implied, but if we're going to be this detailed about documenting build steps then perhaps we should add a `source devel/setup.bash` as well?

gavanderhoorn replied:
And a `source /opt/ros/kinetic/setup.bash`?

AustinDeric replied:
I'd prefer to end the build instructions at a successful build. An examples section should start where the build sections left off and include sourcing the packages.

AustinDeric commented:
In addition to the replied fixes, i removed c++11 flags where they were not needed.  I also added more emphasis in the readme that "build from sources"  is not recommended.

AustinDeric committed:
remove all reference to source build and rebase to indigo devel.

AustinDeric commented:
I took out all references and code related to building from source. @Levi-Armstrong and @gavanderhoorn are we good with the state of this package?  If so, can we create a new branch with it?

Thanks!

Levi-Armstrong commented:
@AustinDeric I created a kinetic-devel branch can you verify that this PR with the new branch on kinetic?

AustinDeric commented:
@Levi-Armstrong i rebased this PR using the [ros-industrial/abb/tree/kinetic-devel](https://github.com/ros-industrial/abb/tree/kinetic-devel) branch.

Everything looks good.  Based on the comments in this PR, i will update the [abb_experimental ](https://github.com/ros-industrial/abb_experimental) kinetic-devel PR so we dont have to repeat ourselves.  Can you get the kinetic-devel branch out?

Thank you all for your hard work!
